pkg/cmd/release/create/create_test.go

```diff
@@ -120,65 +123,242 @@ func TestReleaseCreate_AskQuestions(t *testing.T) {
 		assert.Equal(t, "9.8.4-prerelease", options.Version)
 	})
+}
 
-	t.Run("standard process asking for everything (no package versions) in CaC project", func(t *testing.T) {
-		api := testutil.NewMockHttpServer()
-		qa := testutil.NewAskMocker()
+func TestReleaseCreate_AskQuestions_VersionControlledProject(t *testing.T) {
+	rootResource := testutil.NewRootResource()
 
-		cacProjectID := "Projects-87"
-		repoUrl, _ := url.Parse("http://server/repo.git")
+	const spaceID = "Spaces-1"
 
-		cacProject := fixtures.NewProject(spaceID, cacProjectID, "CaC Project", "Lifecycles-1", "ProjectGroups-1", depProcess.ID)
-		cacProject.PersistenceSettings = projects.NewGitPersistenceSettings(
-			".octopus",
-			projects.NewAnonymousGitCredential(),
-			"main",
-			repoUrl,
-		)
+	projectID := "Projects-87"
+	depProcessDevelopBranch := fixtures.NewDeploymentProcessForVersionControlledProject(spaceID, projectID, "develop")
+
+	depSettings := fixtures.NewDeploymentSettingsForProject(spaceID, projectID, &projects.VersioningStrategy{
+		Template: "#{Octopus.Version.LastMajor}.#{Octopus.Version.LastMinor}.#{Octopus.Version.NextPatch}", // bog standard
+	})
+	depTemplate := &deployments.DeploymentProcessTemplate{NextVersionIncrement: "27.9.3"}
+
+	project := fixtures.NewVersionControlledProject(spaceID, projectID, "CaC Project", "Lifecycles-1", "ProjectGroups-1", depProcessDevelopBranch.ID)
+
+	defaultChannel := fixtures.NewChannel(spaceID, "Channels-34", "CaC Project Default Channel", projectID)
+	altChannel := fixtures.NewChannel(spaceID, "Channels-97", "CaC Project Alt Channel", projectID)
+
+	t.Run("standard process asking for everything (no package versions); specific git commit not set", func(t *testing.T) {
+		api, qa := testutil.NewMockServerAndAsker()
 
 		options := &executor.TaskOptionsCreateRelease{}
 
 		errReceiver := testutil.GoBegin(func() error {
-			// NewClient makes network calls so we have to run it in the goroutine
+			defer testutil.Close(api, qa)
 			octopus, _ := octopusApiClient.NewClient(testutil.NewMockHttpClientWithTransport(api), serverUrl, placeholderApiKey, "")
-			return create.AskQuestions(octopus, qa.AsAsker(), options)
+			return create.AskQuestions(octopus, qa.AsAsker(), spinner, options)
 		})
 
 		api.ExpectRequest(t, "GET", "/api").RespondWith(rootResource)
 
-		api.ExpectRequest(t, "GET", "/api/Spaces-1/projects/all").RespondWith([]*projects.Project{fireProject})
+		api.ExpectRequest(t, "GET", "/api/Spaces-1/projects/all").RespondWith([]*projects.Project{project})
 
 		qa.ExpectQuestion(t, &survey.Select{
 			Message: "Select the project in which the release will be created",
-			Options: []string{"Fire Project"},
-		}).AnswerWith("Fire Project")
+			Options: []string{project.Name},
+		}).AnswerWith(project.Name)
+
+		// CLI will load all the branches and tags
+		api.ExpectRequest(t, "GET", "/api/Spaces-1/projects/"+projectID+"/git/branches").RespondWith(resources.Resources[*projects.GitReference]{
```

Comment:
for discussion: FYI @slewis74 - our unit tests are currently coupled to the internal workings of the API client, which is not great